### PR TITLE
Also test against java 15

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
I want to make sure things are fine on java 11 since that's the version I have installed locally.

### CODE REVIEW

**Comment by olleolleolle** [comment]
My .zshrc locally, needed 

    export JAVA_OPTS="$JAVA_OPTS --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.base/java.io=ALL-UNNAMED"

...perhaps we need such details, too?

**Comment by deivid-rodriguez** [comment]
The `rake integration` task is running into the classic java 11 specific error:

>  (OpenSSL::X509::StoreError) setting default path failed: No password supplied for PKCS#12 KeyStore.

Is that configuration to fix that?

**Comment by deivid-rodriguez** [comment]
Mmmmm, there are also some  failures on the rake 11 job... I'll rerun to see whether they are flakies or not.

**Comment by deivid-rodriguez** [comment]
Alright, so rake 11 failed (might be flaky or not, not yet sure), and rake 10 is hanging, so... Let me propose my solution for everything: drop support for rake 10 and 11 :laughing:. What do you think?

**Comment by olleolleolle** [comment]
@deivid-rodriguez Good, ambitious pace!

**Comment by deivid-rodriguez** [comment]
Flakies were independent from `rake`, they do seem related to the java version being used.

**Comment by deivid-rodriguez** [comment]
Funny we get these warnings on java 16:

> WARN FilenoUtil : Native subprocess control requires open access to the JDK IO subsystem\nPass '--add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.base/java.io=ALL-UNNAMED' to enable.\n

I'll add your suggestion.

**Comment by olleolleolle** [comment]
Oof java 8 complained https://github.com/jruby/warbler/runs/2801902936#step:4:19

```
  /home/runner/.rubies/jruby-9.2.18.0/bin/gem install bundler -v 1.17.3 --no-document
  Unrecognized option: --add-opens
  Error: Could not create the Java Virtual Machine.
  Error: A fatal exception has occurred. Program will exit.
```

**Comment by deivid-rodriguez** [comment]
Yeah, java 8 does not support `--add-opens`. java 11 seems flaky somehow. And java 15 and 16 seem fine. Should we test against java 15 and 16?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,15 +10,15 @@ on:
 
 env:
   JRUBY_OPTS: -J-Xmx896M
-  JAVA_OPTS: --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.base/java.io=ALL-UNNAMED
+  JAVA_OPTS: '-add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.base/java.io=ALL-UNNAMED'
```
Comment (olleolleolle) [inline .github/workflows/ci.yml:13]: Looks like a missing `-` here.

Comment (deivid-rodriguez) [reply .github/workflows/ci.yml:13]: Shit, thanks!

**Comment by deivid-rodriguez** [comment]
I don't know why the options are not taking effect. It seems like the only java consistenty green is java 15. Should we just switch java 8 with java 15? My main motivation here is to get a local enviromnent that matches CI but I'd rather upgrade my java version that downgrade it to achieve that.

**Comment by deivid-rodriguez** [comment]
I went with my last proposal in the end.

**Comment by olleolleolle** [comment]
@deivid-rodriguez My friend was like "think of LTS"

https://www.oracle.com/java/technologies/java-se-support-roadmap.html

> Release 	GA Date 	Premier Support Until 	Extended Support Until 	Sustaining Support
> 11 (LTS) 	September 2018 	September 2023 	September 2026 	Indefinite (edited) 


> If you want to prune, I’d keep the LTS releases (8, 11 now and 17 in September) and only keep the latest non-LTS. So 16 right now.

But I lean towards "if 11 can't build, we don't have the bandwidth to fix that, now, toss it".

**Comment by deivid-rodriguez** [comment]
I see, maybe test on java 8 and 15 for now, and update that to 8 and 17, or maybe just 17 when it's released and we can make it pass?

**Comment by olleolleolle** [comment]
@deivid-rodriguez That sounds prudent, and will probably work!

I hope to create some more clarity around "what is the golden matrix we want to strive for".

**Comment by deivid-rodriguez** [comment]
I think the latest LTS and the latest non-LTS is a good goal. For now the second to last LTS, and the second to last non-LTS seems like a start :)

**Pull request merged**